cmd/svcinit/main.go

```diff
@@ -257,7 +272,7 @@ func main() {
 			must(err)
 
 			// TODO(zbarsky): what is the right behavior here when services are crashing in ibazel mode?
-			criticalPath, _, err = r.UpdateSpecsAndRestart(serviceSpecs, []byte(ibazelCmd))
+			criticalPath, err = r.UpdateSpecsAndRestart(serviceSpecs, servicesErrCh, []byte(ibazelCmd))
```

Comment:
I think you may need to drain the channel before it can be reused, otherwise you could potentially deadlock? I haven't thought this part through fully yet, hence the comment :)